prometheus/src/main/java/org/opensearch/sql/prometheus/storage/model/PrometheusResponseFieldNames.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.prometheus.storage.model;
+
+import static org.opensearch.sql.data.type.ExprCoreType.DOUBLE;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.TIMESTAMP;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.VALUE;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.sql.data.type.ExprType;
+
+
+@Getter
+@Setter
+public class PrometheusResponseFieldNames {
```

Comment:
As I understand, this carries the metadata of Prometheus response, right? If so, because we represent dynamic Prometheus query as a `Table`, I think this should be removed and the logic should be moved to `PrometheusMetricTable.getFieldTypes()` ?

Reply:
The above code is specifically to handle stats output field names which cannot be decided using getFieldTypes.

In the current state.
Logical project decides the field names in the response schema. It analyzes stat expressions and other expressions to come up with the field names in output response. In case of opensearch, the field name matches with the output response from the opensearch. But prometheus doesn't have any field names associated.

For eg:
source = x | stats avg(@value) by handler;
Opensearch produces output with `avg(@value)` as field name for average.
Prometheus we need to map the output to the corresponding key.

Also for dynamic prometheus query, there is no easy way to figure out the output structure in prior without running the query. 

So prometheusMetricTable.getFieldTypes() doesn't work in case of stats and dynamic queries. [Will try to address this after 2.4]